test/unit/cmap/handshake/client_metadata.test.ts

```diff
@@ -1,149 +1,504 @@
-        driver: {
-          name: 'nodejs',
-          version: `${NODE_DRIVER_VERSION}|myVersion`
-        },
-        os: {
-          type: os.type(),
-          name: process.platform,
-          architecture: process.arch,
-          version: os.release()
-        },
-        platform: `Node.js ${process.version}, ${os.endianness()}`
+    context('when there is data from multiple cloud providers in the env', () => {
+      context('unrelated environments', () => {
+        before(() => {
+          // aws
+          process.env.AWS_EXECUTION_ENV = 'non-empty-string';
+          // azure
+          process.env.FUNCTIONS_WORKER_RUNTIME = 'non-empty-string';
+        });
+        after(() => {
+          delete process.env.AWS_EXECUTION_ENV;
+          delete process.env.FUNCTIONS_WORKER_RUNTIME;
+        });
+        it('returns null', () => {
+          expect(getFAASEnv()).to.be.null;
+        });
+      });
+
+      context('vercel and aws which share env variables', () => {
+        before(() => {
+          // vercel
+          process.env.VERCEL = 'non-empty-string';
+          // aws
+          process.env.AWS_EXECUTION_ENV = 'non-empty-string';
+          process.env.AWS_LAMBDA_RUNTIME_API = 'non-empty-string';
+        });
+        after(() => {
+          delete process.env.VERCEL;
+          delete process.env.AWS_EXECUTION_ENV;
+          delete process.env.AWS_LAMBDA_RUNTIME_API;
+        });
+
+        it('parses vercel', () => {
+          expect(getFAASEnv()?.get('name')).to.equal('vercel');
+        });
       });
     });
   });
 
-  context('when no custom driverInfo is provided', () => {
-    const metadata = makeClientMetadata({ driverInfo: {} });
+  describe('makeClientMetadata()', () => {
+    context('when no FAAS environment is detected', () => {
+      it('does not append FAAS metadata', () => {
+        const metadata = makeClientMetadata({ driverInfo: {} });
+        expect(metadata).not.to.have.property(
+          'env',
+          'faas metadata applied in a non-faas environment'
+        );
+        expect(metadata).to.deep.equal({
+          driver: {
+            name: 'nodejs',
+            version: NODE_DRIVER_VERSION
+          },
+          os: {
+            type: os.type(),
+            name: process.platform,
+            architecture: process.arch,
+            version: os.release()
+          },
+          platform: `Node.js ${process.version}, ${os.endianness()}`
+        });
+      });
+    });
 
-    it('does not append the driver info to the metadata', () => {
-      expect(metadata).to.deep.equal({
-        driver: {
-          name: 'nodejs',
-          version: NODE_DRIVER_VERSION
-        },
-        os: {
-          type: os.type(),
-          name: process.platform,
-          architecture: process.arch,
-          version: os.release()
-        },
-        platform: `Node.js ${process.version}, ${os.endianness()}`
+    context('when driverInfo is too large', () => {
```

Comment:
I can live with this, but it would be more clear if you just put the error cases with the appropriate properties where I left the original set of comments. That way it doesn't read "when the driverInfo is too large, throws a [single] error covering [all of] name, version, and platform".